Feature/postgresql 9.5

CloudNiner opened the pull request:
Notes are in the first commit's log.

CloudNiner committed:
Updates the vagrant-cartodb configuration for postgresql 9.5

Also does the following:
- Properly templates ansible group var `postgresql_port` so that
  the value is respected everywhere in the app
- Properly templates postgresql_version so that the vm provisions
  with the correct version. Supported for a value of '9.5'. '9.3'
  may also work.
- Updates the app_config.yml and database.yml files to be generated
  from templates
- Removes the schema_triggers.so configuration from cartodb.postgresql
  role, it is unneccessary as of:
  https://github.com/CartoDB/cartodb/commit/3a199494613fb6cd5670acad0f380bf24433afc0
- Adds ability to configure the common_data settings from group_vars/all
- Adds ability to configure use of cdn static assets from group_vars/all (default false)

These changes require cartodb version 3.12.3 or higher due to the
removal of the postgresql schema triggers dependency. If you need
to develop on an older version of cartodb, use v0.1.0 of this repo.

To upgrade:

If provisioning a new vm:
1. Copy deployment/ansible/group_vars/all.example to
   deployment/ansible/group_vars/all and run `vagrant up`

If provisioning an existing vm (UNTESTED):
Update the following vars in deployment/ansible/group_vars/all:
 - postgresql_version to `9.5`
 - postgresql_port to `5433`

Copy the additional vars in the `# cartodb app_config.yml` section of group_vars/all.example
to your group_vars/all file.

Run `vagrant provision`. Once the app finishes provisioning, reboot the vm
with `vagrant reload` and restart any appropriate services.

This configuration will leave the old 9.3 cluster running at port 5432
and the new 9.5 one on port 5433. If you don't need to keep your data
around its almost certainly preferable to just destroy your old vm and
follow the 'provisioning a new VM' steps above.

Remove notes about app_config.yml from README
This is now handled during provision by copying an ansible template.

CloudNiner commented:
I'm currently testing my instructions in the commit log for upgrading an existing vm. Provisioning is slow. Will tweak the instructions provided as necessary if things turn sour.

hectcastro commented on ansible/roles/cartodb.postgresql/tasks/postgis.yml:
```diff
@@ -4,8 +4,10 @@
     - libxml2-dev
-    - postgresql-9.3-postgis-2.2
-    - postgresql-9.3-postgis-scripts
+    - liblwgeom-2.1.8
```
This was causing a fresh install to bomb. Do you know why this and `postgis` are needed in addition to the last two packages?

CloudNiner replied:
Interesting. No I do not know. These packages are listed as dependencies on the cartodb docs. I'll reproduce and investigate a bit.

Commit:
Allow cartodb dir and private IP to be set via env in Vagrantfile

Commit:
Update vagrant-cartodb to use master branch for cartodb repos

This change was made since this is intended as a way to quickly
spin up a development box, which makes the master branch the
most likely target.

Now uses the version of cartodb-postgresql bundled as a submodule
in cartodb rather than explicitly pulling down a version separately.

CloudNiner commented:
Updated to use master branch for each cartodb repo, since this is intended as a development box. Was able to provision successfully with the group_vars/all.example file currently on this branch. Let me know if you run into any other issues.

Once merged, I'm going to cut a new release with an updated changelog describing these changes along with backwards compatibility with older versions of cartodb.

hectcastro commented:
👍
